core/src/main/java/org/apache/iceberg/rest/auth/RefreshingAuthManager.java

```diff
@@ -18,73 +18,27 @@
  */
 package org.apache.iceberg.rest.auth;
 
-import java.util.List;
-import java.util.concurrent.Future;
 import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.TimeUnit;
 import javax.annotation.Nullable;
 import org.apache.iceberg.util.ThreadPools;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /**
- * An {@link AuthManager} that provides machinery for refreshing authentication data asynchronously,
- * using a background thread pool.
+ * @deprecated since 1.9.0, will be removed in 1.10.0; use {@link ThreadPools#getAuthRefreshPool()}.
  */
+@Deprecated
```

Comment:
if we're deprecating this class, then I don't think we should be removing functionality here?

Reply:
in fact I believe we should only be deprecating this here without updating the functionality (as that would be a breaking change). Once `RefreshingAuthManager` is removed, we can switch the `OAuth2Manager` to use the new auth refresh pool, but I don't think we can deprecate + switch within the same release

Reply:
what if somebody else from the community is extending `RefreshingAuthManager` and relying on its functionality? We're effectively removing the "refresh" part, which is a breaking change in behavior, even though RevAPI wasn't complaining

Reply:
Sorry I commented before I saw your comment 😅 – I agree that's an issue.

Reply:
> What we can do though, is revert the changes to `RefreshingAuthManager` just in case someone is extending that class.

Yes that's what I was proposing in my earlier comment